src/background/permissions-state-tracker.ts

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { Interpreter } from 'background/interpreter';
+import { BrowserAdapter } from 'common/browser-adapters/browser-adapter';
+import { Message } from 'common/message';
+import { Messages } from 'common/messages';
+
+export class PermissionsStateTracker {
+    constructor(private browserAdapter: BrowserAdapter, private interpreter: Interpreter) {}
+
+    public initialize(): void {
+        this.browserAdapter.addListenerOnPermissionsAdded(this.notifyChange);
+        this.browserAdapter.addListenerOnPermissionsRemoved(this.notifyChange);
+        this.notifyChange();
+    }
+
+    public notifyChange(): void {
+        const allUrlAndFilePermissions: string = '*://*/*';
+
+        // tslint:disable-next-line: no-floating-promises
```

Comment:
The tslint error here is legitmate; it's warning you that there's no guarantee about what will happen if an error is thrown from within the promise. Probably better than suppressing the warning would be to add a `.catch` (or convert to an `async` function with a try/catch) that uses `logger.error` to report any exceptions that occur. Probably we should also consider an exception to be the same as a "false" result for the purposes of sending the state update message.

Reply:
Thanks for the suggestions! I like it much more than what I originally had.